ethcore/machine/src/executive.rs

```diff
@@ -350,6 +359,11 @@ impl<'a> CallCreateExecutive<'a> {
 				| Err(vm::Error::OutOfBounds)
 				| Err(vm::Error::Reverted)
 				| Ok(FinalizationResult { apply_state: false, .. }) => {
+					if let Some(addr) = UNPRUNABLE_PRECOMPILE_ADDRESS {
```

Comment:
I think we should have a comment explaining what is going on here.

Reply:
It is commented in UNPRUNABLE_PRECOMPILE_ADDRESS

Reply:
I think it would be good to have something along the lines of "Some state tests written with other implementations in mind need precompile 3 to be left alone". Just to save the reader some time to go out and understand a quite complex issue that very likely is not interesting to them.